src/hubs/neynar.ts

```diff
@@ -6,12 +12,46 @@ export type NeynarHubParameters = {
 
 export const neynar = createHub((parameters: NeynarHubParameters) => {
   const { apiKey } = parameters
+
+  const verifyFrame = async ({
+    frameUrl,
+    trustedData,
+    url,
+  }: VerifyFrameParameters) => {
+    const bytes = hexToBytes(`0x${trustedData.messageBytes}`)
+    const response = await fetch(
+      'https://api.neynar.com/v2/farcaster/frame/validate',
+      {
+        method: 'POST',
+        headers: {
+          accept: 'application json',
+          api_key: apiKey,
+          'content-type': 'application/json',
+        },
+        body: JSON.stringify({
+          message_bytes_in_hex: `0x${trustedData.messageBytes}`,
+        }),
+      },
+    ).then(async (res) => res.json())
+
+    if (!response.valid)
+      throw new Error(`message is invalid. ${response.message}`)
+
+    if (new URL(url).origin !== new URL(frameUrl).origin)
+      throw new Error(`Invalid frame url: ${frameUrl}. Expected: ${url}.`)
+
+    const message = Message.fromBinary(bytes)
+    const frameData = messageToFrameData(message)
+    return { frameData }
+  }
```

Comment:
Maybe we define this in the return statement so we can infer the parameter type instead of explicitly casting:

```tsx
return {
  // ...
  async verifyFrame({ frameUrl, trustedData, url }) {
    // ...
  }
}
```